src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,369 @@
+        // we are now in between intake and drum
+        intakeState = IntakeState.IN_BETWEEN_INTAKE_AND_DRUM;
+
+        // set the timeout here
+        intakeAndDrumTimeout = 0;
+      }
+    } else if (intakeState == IntakeState.IN_BETWEEN_INTAKE_AND_DRUM) {
+      //  the only possible next step is that we are only in the drum
+      //  add a timeout so that if we are here for too long we go back to being empty
+      intakeAndDrumTimeout++;
+
+      if (inputs.isDrumIRBlocked) {
+        intakeState = IntakeState.NOTE_IN_DRUM;
+        this.transitionGamePiece();
+      } else if (intakeAndDrumTimeout >= IntakeConstants.IN_BETWEEN_TIMEOUT_SECONDS * 50) {
+        intakeState = IntakeState.EMPTY;
+        leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_INTAKE_AND_DRUM) {
+      // the only possible next step is that we are only in the drum
+      //    1. this is when the intake IR stops being detected
+      // assuming that we do not need to call transitionGamePiece() again because we already called
+      if (!(inputs.isLeftRollerIRBlocked || inputs.isRightRollerIRBlocked)) {
+        intakeState = IntakeState.NOTE_IN_DRUM;
+        this.repelGamePiece();
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_DRUM) {
+      // the only possible next step is becoming empty between the drum and the kicker
+      if (!inputs.isDrumIRBlocked) {
+        intakeState = IntakeState.IN_BETWEEN_DRUM_AND_KICKER;
+        this.turnTransitionOff();
+        this.repelGamePiece();
+      }
+    } else if (intakeState == IntakeState.NOTE_IN_KICKER) {
+      // the only possible next step is that we become empty after shooting
+      if (!inputs.isKickerIRBlocked) {
+        intakeState = IntakeState.EMPTY;
+        leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+        this.intakeGamePiece();
+      }
+    }
+  }
+
+  public Command getSystemCheckCommand() {
+    /*
+     * Order of what will be tested
+     * 1. make sure everything works at right velocities
+     * 2. then check that all the states it goes through when it gets a note are correct  
+     *
+     */
+    return Commands.sequence(
+      getVelocitiesCheckCommand(),
+      getStatesCheckCommand()
+    ).until(() -> !FaultReporter.getInstance().getFaults(SUBSYSTEM_NAME).isEmpty())
+    .andThen(Commands.runOnce(this::turnIntakeOff))
+    .withName(SUBSYSTEM_NAME + "SystemCheck");
+  }
+
+
+  private Command getVelocitiesCheckCommand() {
+    return Commands.parallel(
+      Commands.run(this::runIntakeStateMachine),
+      Commands.waitSeconds(1).andThen(
+        Commands.runOnce(() -> {
+          for (int i = 0; i < 4; i++) {
+            checkMotorVelocity(intakeMotors[i]);
+          }
+        })
+      )
+    );
+  }
+
+  private void checkMotorVelocity(IntakeMotor motor) {
+    if (motor == IntakeMotor.RIGHT_ROLLER) {
+      if (inputs.rightRollerVelocityRotationsPerSecond 
```

Comment:
We should check if the motor is too slow or too fast. Perhaps something like this:
```
 if (Math.abs(inputs.rightRollerVelocityRotationsPerSecond - inputs.rightRollerReferenceVelocityRPS)
          > IntakeConstants.ROLLER_VELOCITY_TOLERANCE)
```